### ISSUE TITLE
No error when importing `typing.type_check_only`

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#11592]
```py
from typing import type_check_only
```
**Success: no issues found in 1 source file**

**Comment by sobolevn** [comment]
Can you please explain why do you think this should be an error? 
Do you mean older versions of python that don't have this in typing?

**Comment by KotlinIsland** [comment]
https://docs.python.org/3/library/typing.html#typing.type_check_only
> This decorator is itself not available at runtime.

It doesn't actually exist.

```py
> python test.py
Traceback (most recent call last):
  File "/tmp/test.py", line 1, in <module>
    from typing import type_check_only
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name 'type_check_only' from 'typing' (/usr/lib/python3.11/typing.py)
```

**Comment by AlexWaygood** [comment]
We want things like this to be legal, though:

```python 
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import type_check_only
```

This will not fail at runtime, so a blanket "mypy should raise an error if you ever try to import `type_check_only` in a non-stub file" rule wouldn't be good enough.

**Comment by KotlinIsland** [comment]
The docs specify that:
> It is mainly intended to mark classes that are defined in type stub files if an implementation returns an instance of a private class

And if currently using it in a "runtime" context is a huge false negative, wouldn't it be better to always error in used in a py file and when you do want to use put a type ignore on it.

Maybe also allowing it in explicit `TYPE_CHECKING` blocks

**Comment by KotlinIsland** [comment]
related: #9531

**Comment by hauntsaninja** [comment]
Duplicate of https://github.com/python/mypy/issues/9531
